sklearn/feature_extraction/text.py

```diff
@@ -1091,6 +1091,9 @@ def fit(self, raw_documents, y=None):
         -------
         self
         """
+        if self.analyzer != 'word':
+            warnings.warn("The parameter 'stop_words' will not be used"
+                          " since analyzer != 'word'")
```

Comment:
This should only be raised if `stop_words is not None`

Reply:
done